source/mir/random/discrete.d

```diff
@@ -68,14 +68,8 @@ struct Discrete(T)
     this(const(T)[] cdPoints)
     {
         import std.range : assumeSorted;
-        _cdPoints = cdPoints;
-        _cdSortedRange = _cdPoints.assumeSorted!"a <= b";
-    }
-
-    /// cumulative density points
-    const(T)[] cdPoints() const @property
-    {
-        return _cdPoints;
+        _cdSum = cdPoints[$ - 1];
+        _cdSortedRange = cdPoints.assumeSorted!"a <= b";
```

Comment:
Yea, this also works.  It's still a workaround, though ;-)

OTOH it might have some application if e.g. you wanted to generalize the constructor to take an arbitrary `SortedRange` as input (which would probably have to be guaranteed to be a forward range, but not _necessarily_ a bidirectional range).